test/Microsoft.Dnx.Tooling.FunctionalTests/DnuRestoreTests.cs

```diff
@@ -98,5 +98,22 @@ public void Restore_NormalizesVersionCasing(DnxSdk sdk)
 
             Assert.Equal(0, result.ExitCode);
         }
+
+        [Theory, TraceTest]
+        [MemberData(nameof(DnxSdks))]
+        public void Restore_OnlyRewriteInvalidLockfiles(DnxSdk sdk)
+        {
+            var solution = TestUtils.GetSolution<DnuRestoreTests>(
+                sdk,
+                "ThreeLockedProjects",
+                appendSolutionNameToTestFolder: true);
+
+            var result = sdk.Dnu.Restore(solution.SourcePath).EnsureSuccess();
+
+            var outputLines = result.StandardOutput.Split('\n');
+            Assert.Contains(outputLines, line => line.Contains("Following lock file") && line.Contains("Project1"));
```

Comment:
I don't understand what that check would validate. As far as I can tell, if an invalid lock file is locked, it's recreated with `--lock`, but if it's unlocked, it's recreated without locking. I did notice that my Project2 lock file wasn't locked after trying that out, though, so I fixed that.